src/odemis/driver/test/orsay_test.py

```diff
@@ -0,0 +1,560 @@
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from __future__ import division
+
+import os
+import unittest
+from time import sleep
+from odemis.driver import orsay
+
+TEST_NOHW = (os.environ.get("TEST_NOHW", 0) != 0)  # Default to Hw testing
+
+CONFIG_PSUS = {"name": "pneumatic-suspension", "role": "pneumatic-suspension"}
+CONFIG_PRESSURE = {"name": "pressure", "role": "chamber"}
+CONFIG_PSYS = {"name": "pumping-system", "role": "pumping-system"}
+CONFIG_UPS = {"name": "ups", "role": "ups"}
+
+CONFIG_ORSAY = {"name": "Orsay", "role": "orsay", "host": "192.168.56.101",
+                "children": {"pneumatic-suspension": CONFIG_PSUS,
+                             "pressure": CONFIG_PRESSURE,
+                             "pumping-system": CONFIG_PSYS,
+                             "ups": CONFIG_UPS}
+                }
+
+
+class TestOrsayStatic(unittest.TestCase):
+    """
+    Tests which don't need an Orsay component ready
+    """
+
+    def test_creation(self):
+        """
+        Test to create an Orsay component
+        """
+        try:
+            oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            self.fail(e)
+        self.assertEqual(len(oserver.children.value), 4)
+
+        oserver.terminate()
+
+
+class TestOrsay(unittest.TestCase):
+    """
+    Tests to run on the main Orsay component
+    """
+
+    def setUp(self):
+        """
+        Setup the Orsay client
+        """
+        try:
+            self.oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            self.skipTest(e)
+
+        for child in self.oserver.children.value:
+            if child.name == CONFIG_PSUS["name"]:
+                self.psus = child
+            elif child.name == CONFIG_PRESSURE["name"]:
+                self.pressure = child
+            elif child.name == CONFIG_PSYS["name"]:
+                self.psys = child
+            elif child.name == CONFIG_UPS["name"]:
+                self.ups = child
+
+    def tearDown(self):
+        """
+        Terminate the Orsay client
+        """
+        self.oserver.terminate()
+
+    def test_updateProcessInfo(self):
+        """
+        Check that the processInfo VA is updated properly and an exception is raised when the wrong parameter is passed
+        """
+        self.assertRaises(Exception, self.oserver._updateProcessInfo,
+                          {'parameter': self.oserver.datamodel.HybridPlatform.Cancel})
+        if not TEST_NOHW:
+            self.skipTest("TEST_NOHW is not set, cannot force data on Actual parameters of Orsay server outside of "
+                          "simulation")
+        test_string = "Some process information"
+        self.oserver.datamodel.HybridPlatform.ProcessInfo.Actual = test_string
```

Comment:
Instead of always writing `self.oserver.datamodel`, store the `datamodel` directly on the self in the setUpClass()